### PR TITLE
Validate dimensionality

### PR DESCRIPTION
**Pull request by Enet4**
I found a problem while fuzzing the crate. If `dim[0]` contains an inappropriate value, attempting to read a volume would panic. This PR makes some changes to accommodate the fix.

- new error variant for inconsistent dim field values (must be positive, and `dim[0]` must not be higher than 7).
- raise this error if `dim[0]` is incorrect.
- as a bonus, I added public methods to `NiftiHeader`, which validate the values in the process.

### CODE REVIEW

File: src/header.rs [hunk 1]
```diff
@@ -231,6 +231,31 @@ impl NiftiHeader {
...
+    }
+
+    /// Retrieve and validate the number of dimensions of the volume. This is
+    /// `dim[0]` after the necessary byte order conversions.
+    /// 
+    /// # Error
+    /// 
+    /// `NiftiError::` if `dim[0]` does not represent a valid dimensionality.
+    pub fn dimensionality(&self) -> Result <usize> {
+        if self.dim[0] > 7 {
```
Comment (nilgoyette) [inline src/header.rs]: Is 0 a valid dimensionality?

File: src/lib.rs [hunk 2]
```diff
@@ -47,6 +47,7 @@
 //!
 #![deny(missing_debug_implementations)]
 #![warn(missing_docs, unused_extern_crates, trivial_casts, unused_results)]
+#![recursion_limit = "128"]
```
Comment (nilgoyette) [inline src/lib.rs]: Can you please tell me why we need a recursion limit? I don't remember any use of recursion in this lib.

**Comment by Enet4** [comment]
Thanks for reviewing, @nilgoyette. I can double-check those patterns and replace occurrences where reasonable the next few days.

More comments inline.

File: src/header.rs (hunk 1 above)
Comment (Enet4) [reply src/header.rs]: Great question, I don't know. Might be worth looking into this.

File: src/lib.rs (hunk 2 above)
Comment (Enet4) [reply src/lib.rs]: This is the compiler's recursive operation limit. Recursion is used internally by the `quick_error!` macro. Once a significant number of variants are declared, which is the case here, we need to push this value a bit higher.

**Comment by nilgoyette** [comment]
LGTM.

**Pull request merged**